docs/docs/advanced/deploy-eliza-to-tee-with-oasis-rofl.md

```diff
@@ -0,0 +1,257 @@
+
+---
+
+### Prerequisites
+
+Before getting started with Eliza in TEE, ensure you have:
+
+- [Oasis CLI](https://github.com/oasisprotocol/cli/releases) installed
+- Docker Desktop or Orbstack (Orbstack is recommended)
+- Testnet tokens from the [Oasis Faucet](https://faucet.testnet.oasis.dev/)
+- A character file for your Eliza agent (you can use the example character file or create your own)
+
+---
+
+### Environment Setup
+
+To set up your environment for TEE development:
+
+1. **Create a keypair**
+
+    ```bash
+    oasis wallet create <account-name>
+    ```
+
+2. **Request Testnet Tokens**
+
+    Visit the [Oasis Faucet](https://faucet.testnet.oasis.dev/) to request TEST tokens.
+
+3. **Create a ROFL Application**
+
+    Create the ROFL application:
+
+    ```bash
+    oasis rofl create --network testnet --account <account-name>
+    ```
+
+### Build and Deploy the ROFL app
+
+1. **Configure the ROFL app**
+
+    Update the `rofl-compose.yaml` file with your configuration:
+
+    ```yaml
+    services:
+      eliza:
+        restart: always
+        image: ghcr.io/elizaos/eliza:latest
+        build:
+          context: .
+          dockerfile: Dockerfile
+        platform: linux/amd64
+        environment:
+          - CHARACTERS=${CHARACTERS}
+          - SERVER_PORT=${SERVER_PORT}
+          # Add other required environment variables
+    ```
+
+2. **Encrypt Secrets**
+
+    Encrypt all required environment variables. You can use any character file of your choice - here we're using the staff engineer character as an example. Refer to the Secret Management section above for a comprehensive list of secrets you might want to set:
+
+    ```bash
+    # Set the character file path (you can use any character file of your choice)
+    echo -n "characters/staff-engineer.character.json" | oasis rofl secret set CHARACTERS -
+    echo -n "3000" | oasis rofl secret set SERVER_PORT -
+    # Add other required secrets
+    ```
+
+    **WARNING:** Secrets are end-to-end encrypted with your account key and an ephemeral key of the chain where it will be deployed to (Testnet or Mainnet). Migrating encrypted secrets from one deployment to another is not possible.
+
+3. **Configure ROFL Resources**
+
+    Adjust the resource allocation in your `rofl.yaml` file to match your application's requirements. This includes memory, CPU, and storage settings:
+
+    ```yaml
+    resources:
+      memory: 16384  # Memory in MB (16GB in this example)
+      cpus: 1        # Number of CPU cores
+      storage:
+        kind: disk-persistent
+        size: 10240  # Storage size in MB (10GB in this example)
+    ```
+
+    Adjust these values based on your specific needs:
+    - `memory`: Set the amount of RAM your application needs (in MB)
+    - `cpus`: Specify the number of CPU cores to allocate
+    - `storage`: Configure persistent storage with the required size
+
+4. **Build the ROFL Application**
+
+    For MacOS users, use the following Docker image:
```

Comment:
Anyone can use Docker to build ROFL. In fact, it's the preferred way to do it if you need reproducible builds.

```suggestion
    Use the following Docker image for reproducible ROFL builds:
```